src/api/ledger/transaction.js

```diff
@@ -62,11 +62,19 @@ function getTransactionAsync(identifier: string, options: TransactionOptions,
       error = new errors.NotFoundError('Transaction not found');
     }
 
+    const minLedgerVersion = options.minLedgerVersion;
+
     if (error instanceof errors.NotFoundError
-        && !utils.hasCompleteLedgerRange(remote,
-            options.minLedgerVersion, maxLedgerVersion)) {
-      callback(new errors.MissingLedgerHistoryError('Transaction not found,'
-        + ' but the server\'s ledger history is incomplete'));
+        && !utils.hasCompleteLedgerRange(remote, minLedgerVersion,
+                                         maxLedgerVersion)) {
+      if (minLedgerVersion && maxLedgerVersion
+          && utils.isNonContiguousLedgerRange(remote, minLedgerVersion, 
+                                         maxLedgerVersion)) {
+        callback(new errors.NonContiguousLedgerHistoryError());
+      } else if (maxLedgerVersion 
+                 && utils.isPendingLedgerVersion(remote, maxLedgerVersion)) {
+        callback(new errors.PendingLedgerError());
+      }
```

Comment:
`utils.hasCompleteLedgerRange` can be either true or false even if `maxLedgerVersion` is undefined, so it may never reach line 86. Also more generally, you should try to be defensive and avoid this kind of coupling between different branches of an if/else statement. A good rule of thumb is to usually not end with an "else if" because it quite often indicates that there are missing cases.